### PR TITLE
Add Salesforce Field Audit Trail and Setup Audit Trail connectors

### PR DESCRIPTION
**Pull request by brianbeklaviyo**
## Summary
This PR adds two new Salesforce connectors to Grove for collecting audit trail data:
- **Field Audit Trail** (`sf_field_audit_trail`) - Collects field-level change history from the `FieldHistoryArchive` Big Object
- **Setup Audit Trail** (`sf_setup_audit_trail`) - Collects setup and configuration changes from the `SetupAuditTrail` object

### Field Audit Trail Connector (`sf_field_audit_trail`)
- **Purpose**: Collects field-level change history for Salesforce objects
- **Data Source**: `FieldHistoryArchive` Big Object
- **Configuration**: Requires `object_type` parameter (e.g., "Account", "Contact", "Lead")
- **Use Cases**: Compliance auditing, data lineage tracking, change monitoring

### Setup Audit Trail Connector (`sf_setup_audit_trail`)
- **Purpose**: Collects setup and configuration changes in Salesforce
- **Data Source**: `SetupAuditTrail` object
- **Use Cases**: Security auditing, configuration change tracking, compliance monitoring

## Features
- ✅ **Dual Authentication Support**: Both OAuth 2.0 and legacy username/password authentication
- ✅ **Comprehensive Error Handling**: Handles `INVALID_TYPE`, `INVALID_FIELD`, rate limiting, and HTTP errors
- ✅ **Incremental Collection**: Pointer-based collection for efficient data retrieval
- ✅ **Big Object Support**: Proper handling of Salesforce Big Object query limitations
- ✅ **Production Ready**: Includes retry logic, logging, and proper error messages

## Configuration Templates
Added OAuth and legacy authentication templates for both connectors:
- `templates/configuration/salesforce/field_audit_trail_oauth.json`
- `templates/configuration/salesforce/field_audit_trail_legacy.json`
- `templates/configuration/salesforce/setup_audit_trail_oauth.json`
- `templates/configuration/salesforce/setup_audit_trail_legacy.json`

## Testing
- ✅ **Comprehensive Test Coverage**: 34 new test cases covering authentication, data collection, error handling, and edge cases

## Files Added/Modified
- `grove/connectors/sf/field_audit_trail.py` - Field Audit Trail connector
- `grove/connectors/sf/setup_audit_trail.py` - Setup Audit Trail connector
- `tests/test_connectors_sf_field_audit_trail.py` - Field Audit Trail tests
- `tests/test_connectors_sf_setup_audit_trail.py` - Setup Audit Trail tests
- `grove/connectors/sf/__init__.py` - Updated to import new connectors
- `pyproject.toml` - Registered new connectors
- Configuration templates (4 files)

## Requirements
- **Field Audit Trail**: Requires Field Audit Trail to be enabled in Salesforce
- **Setup Audit Trail**: Requires Setup Audit Trail to be enabled in Salesforce
- **Permissions**: User must have appropriate read permissions for audit data

## Usage Example
```json
{
  "client_id": "YOUR_CONNECTED_APP_CLIENT_ID",
  "client_secret": "YOUR_CONNECTED_APP_CLIENT_SECRET",
  "identity": "YOUR_SALESFORCE_USERNAME",
  "instance_url": "https://your-instance.my.salesforce.com",
  "name": "salesforce-field-audit-trail",
  "connector": "sf_field_audit_trail",
  "object_type": "Account",
  "start_date": "2024-01-01T00:00:00.000Z"
}
```

## Testing
All tests pass (70/70 Salesforce connector tests) with comprehensive coverage of:
- Authentication success/failure scenarios
- Data collection with various record types
- Error handling for common Salesforce API errors
- Pointer management and incremental collection
- Configuration validation

Ready for review! This PR adds production-ready Salesforce audit trail connectors with comprehensive testing and documentation.

### CODE REVIEW

File: grove/connectors/sf/field_audit_trail.py [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+                password=self.key,
+                security_token=self.token,
+            )
+            return sf_session, sf_instance
+        except (SalesforceError, requests.exceptions.RequestException) as err:
+            raise RequestFailedException(
+                f"Unable to authenticate with Salesforce using legacy authentication: {err}"
+            )
+
+    def _query_with_retry(self, client: Salesforce, soql_query: str) -> Dict[str, Any]:
```
Comment (tdeslauriers) [inline grove/connectors/sf/field_audit_trail.py]: Also, this query retry logic appears to be duplicative of code in both setup_audit_trail.py line 326 and existing threat_detection.py file, line 362.

Would it be possible to extract this out into a base class?

Comment (brianbeklaviyo) [reply grove/connectors/sf/field_audit_trail.py]: [f124931](https://github.com/hashicorp-forge/grove/pull/117/commits/f12493155407bfd911ed00904a8916dc4b8d1e30)

File: grove/connectors/sf/field_audit_trail.py [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+            return SF_OAUTH_TOKEN_URL_DEFAULT
+
+        # Extract the base domain from the instance URL and construct OAuth endpoint
+        # e.g., https://company.my.salesforce.com -> https://company.my.salesforce.com/services/oauth2/token
+        instance_url = self.instance_url.rstrip("/")
+        oauth_url = f"{instance_url}/services/oauth2/token"
+
+        return oauth_url
+
+    def get_oauth_access_token(self) -> Tuple[str, str]:
```
[the remark this answers is not in the document]
Comment (brianbeklaviyo) [reply grove/connectors/sf/field_audit_trail.py]: [f124931](https://github.com/hashicorp-forge/grove/pull/117/commits/f12493155407bfd911ed00904a8916dc4b8d1e30)
